packages/storage/compat/reference.ts

```diff
@@ -0,0 +1,231 @@
+   */
+  putString(
+    value: string,
+    format: StringFormat = StringFormat.RAW,
+    metadata?: Metadata
+  ): types.UploadTask {
+    validate(
+      'putString',
+      [stringSpec(), stringSpec(formatValidator, true), metadataSpec(true)],
+      arguments
+    );
+    this._throwIfRoot('putString');
+    return new UploadTaskCompat(
+      uploadString(this.delegate, value, format, metadata),
+      this
+    );
+  }
+
+  /**
+   * List all items (files) and prefixes (folders) under this storage reference.
+   *
+   * This is a helper method for calling list() repeatedly until there are
+   * no more results. The default pagination size is 1000.
+   *
+   * Note: The results may not be consistent if objects are changed while this
+   * operation is running.
+   *
+   * Warning: listAll may potentially consume too many resources if there are
+   * too many results.
+   *
+   * @return A Promise that resolves with all the items and prefixes under
+   *  the current storage reference. `prefixes` contains references to
+   *  sub-directories and `items` contains references to objects in this
+   *  folder. `nextPageToken` is never returned.
+   */
+  listAll(): Promise<types.ListResult> {
+    validate('listAll', [], arguments);
+    return listAll(this.delegate).then(
+      r => new ListResultCompat(r, ref => new ReferenceCompat(ref))
+    );
+  }
+
+  /**
+   * List items (files) and prefixes (folders) under this storage reference.
+   *
+   * List API is only available for Firebase Rules Version 2.
+   *
+   * GCS is a key-blob store. Firebase Storage imposes the semantic of '/'
+   * delimited folder structure.
+   * Refer to GCS's List API if you want to learn more.
+   *
+   * To adhere to Firebase Rules's Semantics, Firebase Storage does not
+   * support objects whose paths end with "/" or contain two consecutive
+   * "/"s. Firebase Storage List API will filter these unsupported objects.
+   * list() may fail if there are too many unsupported objects in the bucket.
+   *
+   * @param options See ListOptions for details.
+   * @return A Promise that resolves with the items and prefixes.
+   * `prefixes` contains references to sub-folders and `items`
+   * contains references to objects in this folder. `nextPageToken`
+   * can be used to get the rest of the results.
+   */
+  list(options?: ListOptions | null): Promise<types.ListResult> {
+    validate('list', [listOptionSpec(true)], arguments);
+    return list(this.delegate, options).then(
+      r => new ListResultCompat(r, ref => new ReferenceCompat(ref))
+    );
+  }
+
+  /**
+   * A promise that resolves with the metadata for this object. If this
+   * object doesn't exist or metadata cannot be retreived, the promise is
+   * rejected.
+   */
+  getMetadata(): Promise<Metadata> {
+    validate('getMetadata', [], arguments);
+    return getMetadata(this.delegate);
+  }
+
+  /**
+   * Updates the metadata for this object.
+   * @param metadata The new metadata for the object.
```

Comment:
Just want to mention that we are changing to use tsdoc for documenting in vNext, which is used by api-documenter to generate reference docs. It is similar to jsdoc, but has some differences, for exmaple,
you need to put a `-` after the parameter name and it's `returns` instead of `return`. See https://api-extractor.com/pages/tsdoc/doc_comment_syntax/ for the complete syntax.

You don't have to change the comments for the existing code.